EIPS/eip-3135.md

```diff
@@ -0,0 +1,203 @@
+| newIssuer | The new issuer address |
+
+#### deposit
+
+User can take part of token balance as deposit to ensure micropayments. Token can only be claimed from deposit balance. This works likes 'increaseAllowance', except the deposit is removed from user's balance.
+
+| Parameter | Description |
+| ---------|-------------|
+| amount | amount to deposit |
+
+#### withdraw
+
+This function should have different implementation based on business model. For real-world businesses, such as recharge card of general store or membership card of fitness center, withdraw should be called by merchant, act as refund function to return remaining deposit balance to user's balance; for online businesses, user may found it hard to contact with service provider, so the deposit is called by user itself, with limitations such as epoch number didn't change for some period of time. Also gets token in deposit back.
+
+| Parameter | Description |
+| ---------|-------------|
+| to | the deposit account |
+| amount | amount to return |
+
+#### depositBalanceOf
+
+Returns deposit balance and epoch of user account. The epoch is an increasing number which is added by one when account get claimed or withdraw. Epoch is a part of sign data, verified by both user and provider, prevents double spend and double claim.
+
+| Parameter | Description |
+| ---------|-------------|
+| user | user account address |
+
+#### claim
+
+Service provider (issuer) recovers unsigned message, then checks the message with it's signature (probably using built-in function 'ecrecover'). If the signature is successfully verified, provider get "credit" amount token from user's deposit balance and epoch is increased by 1. It is worth mentioning that the provider may not need to get token back because this token is basically created by this provider and may holds no value to itself. In that case, the claim function can release some other more valuable token to provider. 
+
+| Parameter | Description |
+| ---------|-------------|
+| from | which user account to be claimed |
+| credit | how many token should be claimed |
+| epoch | current epoch number |
+| signature| micropayment message signature signed by user|
+
+
+### Events
+
+#### Deposit
+
+Event emits when user calls deposit function
+
+| Parameter | Description |
+| ---------|-------------|
+| from | indexed msg.sender |
+| amount | deposit amount|
+
+#### Withdraw
+
+| Parameter | Description |
+| ---------|-------------|
+| to | the deposit account |
+| amount | amount to return |
+
+#### TransferIssuer
+
+Event emits when transferIssuer is called
+
+| Parameter | Description |
+| ---------|-------------|
+| oldIssuer | old issuer/beneficiary |
+| newIssuer | new issuer/beneficiary |
+
+#### Claim
+
+Event emits when claim is called
+
+| Parameter | Description |
+| ---------|-------------|
+| from | user account to be claimed |
+| to | msg.sender, the issuer |
+| epoch | which epoch is based on |
+| credit | how many token be claimed |
```

Comment:
Consider using [NatSpec](https://docs.soliditylang.org/en/v0.7.2/natspec-format.html) for inline comments in the interface, rather than a separate set of markdown definitions.  This would make this specification a bit more tool-friendly.